engineyard/ey-cookbooks-dev-v7
# [FBZ-10499] Migrate eydr Recipe to v7

Description of your patch
-------------
Added eydr recipe support for v7 stack using PostgreSQL versions 10, 11, 12, 13 and 14

Recommended Release Notes
-------------
Added eydr recipe support for v7 stack using PostgreSQL versions 10, 11, 12, 13 and 14

Estimated risk
-------------
High

Components involved
-------------
Clients using eydr recipe

Dependencies
-------------
Environments that use either MySQL or PostgreSQL as database

Description of testing done
-------------
- Create 2 v7 environments (with the other on a different region) using any of the PostgreSQL version listed below:
  - 10 (via UI)
  - 11 (via UI)
  - 12 (via Environment Variables and using changes from PR145)
  - 13 (via Environment Variables and using changes from PR145)
  - 14 (via Environment Variables and using changes from PR145)
- Follow the `README.md` file instructions to setup the pre-requisites
- Upload `eydr` custom recipe with the modified details for Master env and Slave env
- Test if replication or failover will work via modifying `eydr/attributues/replication.rb`
- For replication:
  - Once set to `true`, the Slave environment should be streaming replication (can be verified via `select * from pg_stat_wal_receiver;`)
  - Also can be verified from Master env via `select * from pg_stat_replication;`
  - Test by adding data onto Master env and check if Slave env gets the update.

QA Instructions
-------------
- Create 2 v7 environments (with the other on a different region) using any of the PostgreSQL version listed below:
  - 10 (via UI)
  - 11 (via UI)
  - 12 (via Environment Variables and using changes from PR145)
  - 13 (via Environment Variables and using changes from PR145)
  - 14 (via Environment Variables and using changes from PR145)
- Follow the `README.md` file instructions to setup the pre-requisites
- Upload `eydr` custom recipe with the modified details for Master env and Slave env
- Test if replication or failover will work via modifying `eydr/attributues/replication.rb`
- For replication:
  - Once set to `true`, the Slave environment should be streaming replication (can be verified via `select * from pg_stat_wal_receiver;`)
  - Also can be verified from Master env via `select * from pg_stat_replication;`
  - Test by adding data onto Master env and check if Slave env gets the update.
